### PR TITLE
feat(tpu): add tpu queued resources create spot

### PR DESCRIPTION
**Pull request by TetyanaYahodska**
## Description
Documentation - https://cloud.google.com/tpu/docs/spot

Python sample -> https://github.com/GoogleCloudPlatform/python-docs-samples/pull/12716/files#diff-75c744e8e8de9610368a82581c10456cb996327510d4cf5f3a6bdefe898a073b
Fixes #<ISSUE-NUMBER>

Note: Before submitting a pull request, please open an issue for discussion if you are not associated with Google.

## Checklist

- [x] I have followed [Sample Format Guide](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/SAMPLE_FORMAT.md)
- [ ] `pom.xml` parent set to latest `shared-configuration`
- [ ] Appropriate changes to README are included in PR
- [ ] These samples need a new **API enabled** in testing projects to pass (let us know which ones)
- [ ] These samples need a new/updated **env vars** in testing projects set to pass (let us know which ones)
- [x] **Tests** pass:   `mvn clean verify` **required**
- [x] **Lint**  passes: `mvn -P lint checkstyle:check` **required**
- [ ] **Static Analysis**:  `mvn -P lint clean compile pmd:cpd-check spotbugs:check` **advisory only**
- [ ] This sample adds a new sample directory, and I updated the [CODEOWNERS file](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/.github/CODEOWNERS) with the codeowners for this sample
- [ ] This sample adds a new **Product API**, and I updated the [Blunderbuss issue/PR auto-assigner](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/.github/blunderbuss.yml) with the codeowners for this sample 
- [ ] Please **merge** this PR for me once it is approved

### CODE REVIEW

File: tpu/src/main/java/tpu/CreateSpotQueuedResource.java
Comment (minherz) [inline tpu/src/main/java/tpu/CreateSpotQueuedResource.java:76]: We want to keep the code consistent. Consider to replace with

```suggestion
      var parent = String.format("projects/%s/locations/%s", projectId, zone);
      var resourceName = String.format("projects/%s/locations/%s/queuedResources/%s",
                      projectId, zone, queuedResourceId);
      SchedulingConfig schedulingConfig = SchedulingConfig.newBuilder()
          .setPreemptible(true)
          .build();

      Node node =
          Node.newBuilder()
              .setName(nodeName)
              .setAcceleratorType(tpuType)
              .setRuntimeVersion(tpuSoftwareVersion)
              .setSchedulingConfig(schedulingConfig)
              .setQueuedResource(resourceName)
```

File: tpu/src/test/java/tpu/QueuedResourceIT.java
Comment (minherz) [inline tpu/src/test/java/tpu/QueuedResourceIT.java:142]: nit: consider giving it more verbose names. It is easy to confuse between `mockedTpuClient` and `mockTpuClient`.

For example, the latter can be called `mockedClientInstance`

**Review by minherz — APPROVED** [review]
lgtm. I have left a couple of comments. Consider to apply them before merging.

If the PR branch is out of sync with main, DO NOT merge changes from the main branch. Use "rebase".

File: tpu/src/main/java/tpu/CreateSpotQueuedResource.java
Comment (TetyanaYahodska) [reply tpu/src/main/java/tpu/CreateSpotQueuedResource.java:76]: Thank you! Fixed

File: tpu/src/test/java/tpu/QueuedResourceIT.java
Comment (TetyanaYahodska) [reply tpu/src/test/java/tpu/QueuedResourceIT.java:142]: Thank you! Changed name

**Pull request merged**
